scripts/utils/starknet.py

```diff
@@ -424,6 +472,14 @@ async def call(contract_name, function_name, *inputs, address=None):
     return await contract.functions[function_name].call(*inputs)
 
 
+async def call(contract, *args, **kwargs):
+    return await (
+        call_address(contract, *args, **kwargs)
+        if isinstance(contract, int)
+        else call_contract(contract, *args, **kwargs)
+    )
```

Comment:
Just to see if I understand this right, if the deployments exist, we load the abi and call from `Contract` object, otherwise we take the address and do a raw call using `call_contract`?

Reply:
I think it's the opposite:
```
 call_address(contract, *args, **kwargs)
        if isinstance(contract, int)
```

means -> do a raw call if `contract` is an integer, i.e. not a contract name (string), as integers represent addresses in cairo
-> else do a contract call

Reply:
the naming is a bit misleading -> 
could be commented above that contract (the first var) is either a contract name or a contract address

Reply:
Ah sorry I meant a lower level `call_contract` https://github.com/kkrt-labs/kakarot/pull/659/files#diff-0da6276ff678d86aaa4359903ad0e57bbd931de7cda2d03ba561a58592bc6236R455, but yeah agree with the flow you illustrated 👍